### PR TITLE
feat: Filter notifications on monitor for APKAM

### LINKED ISSUE #1444: APKAM - access changes for monitor verb
### Is your feature request related to a problem? Please describe.

When user is authenticated via APKAM, allow only the keys with authorized namespaces to flow through the verbs.

- Monitor verb: Allow only the notification where keys with authorized namespaces to stream through the monitor connection.
- Notify verb: Allow only the notification where keys with authorized namespaces to be processed through the notify verb.

### Describe the solution you'd like

- Refer to the implementation https://github.com/atsign-foundation/at_server/pull/1443
- In monitor verb handler, add a check to verify if the namespaces of a key are authorized with enrollmentId used to authenticate.
- Similar to monitor verb, add a check in notify verb check for authorization of namespace.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
- Add APKAM functionality to monitor verb. 
- If the connection is authenticate via APKAM, send notifications only if enrollment namespaces contains .* or __manage or enrollment is authorized for the key's namespace.
- Refactor the monitor_verb_hander.dart 
- Add unit tests for monitor verb.
- Reduce the log-level in `_sendNotificationByEnrollmentNamespaceAccess` method because it gets invoked for every notification causing server logs more noisy.

**- How I did it**
- Introduce method `_sendNotificationByEnrollmentNamespaceAccess` which verify if the enrollment is authorized to send notifications when connection is authenticated via APKAM
- The existing logic is moved to `_sendLegacyNotification` which gets invoked when the connection is authenticated via PKAM verb.
- Change the name of the method `processReceivedNotification` to `_sendNotificationToClient` to better reflect the purpose of the method.

**- How to verify it**
- All the tests should pass.

**- Description for the changelog**
- Add APKAM functionality to monitor verb.

### CODE REVIEW

File: packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart [hunk 1]
```diff
@@ -36,74 +40,144 @@ class MonitorVerbHandler extends AbstractVerbHandler {
...
+      return;
+    }
+    if (enrollDataStoreValue.namespaces.isEmpty) {
+      logger.info('No namespaces are enrolled for the enrollmentId:'
+          ' ${(atConnection.getMetaData() as InboundConnectionMetadata).enrollmentId}');
+      return;
+    }
+    // separate namespace from the notification key
+    String namespaceFromKey = notification.notification!
+        .substring(notification.notification!.indexOf('.') + 1);
```
Comment (gkc) [inline packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart:157]: Need to handle situation where there is no namespace

**Review by gkc — CHANGES_REQUESTED** [review]
Left a comment

File: packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart (hunk 1 above)
Comment (sitaram-kalluri) [reply packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart:157]: Added a check before this line to check if the `notification.notification` contains ".". If not available, it indicates the namespace is not present and do nothing.

File: packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart [hunk 2]
```diff
@@ -13,7 +17,7 @@ class MonitorVerbHandler extends AbstractVerbHandler {
 
   late InboundConnection atConnection;
 
-  String? regex;
+  late String regex;
```
Comment (gkc) [inline packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart:20]: neither the inbound connection nor the regex should be instance variables

**Review by gkc — CHANGES_REQUESTED** [review]
neither the inbound connection nor the regex should be instance variables in MonitorVerbHandler

**Comment by sitaram-kalluri** [comment]
> neither the inbound connection nor the regex should be instance variables in MonitorVerbHandler

Removed atConnection and regex as instance variables in the monitor_verb_handler.dart. The corresponding changes in at_persistence_secondary_server are in the following PR: https://github.com/atsign-foundation/at_server/pull/1531

File: packages/at_secondary_server/lib/src/connection/inbound/inbound_connection_metadata.dart [hunk 3]
```diff
@@ -26,4 +26,8 @@ class InboundConnectionMetadata extends AtConnectionMetaData {
 
   /// A unique identifier generated for a client's APKAM enroll request
   String? enrollmentId;
+
+  /// This is used by the connections where monitor is set to true to filter notifications
+  /// based on the regular expression.
+  String regex = '.*';
```
Comment (gkc) [inline packages/at_secondary_server/lib/src/connection/inbound/inbound_connection_metadata.dart]: I don't believe we should have this here. Let's discuss.

**Comment by sitaram-kalluri** [comment]
The changes are deferred. Hence closing the PR.

File: packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart (hunk 2 above)
Comment (sitaram-kalluri) [reply packages/at_secondary_server/lib/src/verb/handler/monitor_verb_handler.dart:20]: As discussed, the changes are deferred

File: packages/at_secondary_server/lib/src/connection/inbound/inbound_connection_metadata.dart (hunk 3 above)
Comment (sitaram-kalluri) [reply packages/at_secondary_server/lib/src/connection/inbound/inbound_connection_metadata.dart]: Removed this change.

**Review by gkc — APPROVED** [review]

**Pull request merged**